### DISCUSSION TITLE
How to apply constant velocity on blocks

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#24640] [Q&A Modules: Solid mechanics] [answered]
<img width="937" alt="Screenshot 2023-06-08 at 9 52 16 PM" src="https://github.com/idaholab/moose/assets/93891863/77ef7877-18ea-4697-bcb9-a6cd8d0f3854">
This might be an easy question but I'm struggling to find an appropriate way to do it: I would like apply constant velocity on two blocks starting as initial condition shown in the figure. That is, I would like to have constant displacement increment keep adding into the system as time marching. How should I achieve this? Thanks in advance.

**Comment by Rahim-Habibi** [comment]
This probably can help you :
 https://mooseframework.inl.gov/source/bcs/PresetVelocity.html
or:
https://mooseframework.inl.gov/source/auxkernels/NewmarkVelAux.html

**Comment by GiudGiud** [comment]
I think you want to displace the two blocks right?
@recuero

You ll want to use the Dynamics tensor mechanics action.
I could not find its documentation.
Please look for examples in the repo while we locate it

```
    [./TensorMechanics/DynamicMaster]
      [all]
        add_variables = true
        strain = FINITE
        use_automatic_differentiation = false
        generate_output = 'stress_xx stress_xy stress_xz stress_yy stress_yz stress_zz
        strain_xx strain_xy strain_xz strain_yx strain_yy strain_yz strain_zx
        strain_zy strain_zz
  elastic_strain_xx  elastic_strain_xy  elastic_strain_xz  elastic_strain_yx  elastic_strain_yy
   elastic_strain_yz  elastic_strain_zx  elastic_strain_zy  elastic_strain_zz  mechanical_strain_xx
    mechanical_strain_xy  mechanical_strain_xz  mechanical_strain_yx  mechanical_strain_yy
     mechanical_strain_yz  mechanical_strain_zx  mechanical_strain_zy  mechanical_strain_zz
         vonmises_stress'
      []
    []
  []
```

looks like this

**Reply by recuero** [reply]
If @chunhuizhao478 is using explicit, the Newmark way may not work in this case. The most immediate thing that comes to mind is to use a function that depends linearly on time (`FunctionDirichletBC`) applied to the displacements, have you tried that?

**Reply by chunhuizhao478** [reply]
Thanks Antonio, I indeed tried this, however, I think BC system don't have Block-Restrict functionality, I may only apply bc along the boundary at the interface

**Reply by GiudGiud** [reply]
Hello

The idea is that you apply the displacement on the boundary, and the rigidity of the solid makes it so that the entire solid moves.
So you dont need to "block restrict"

Guillaume

**Reply by hugary1995** [reply]
The OP's description is unclear. I think he wants to apply an initial velocity to the entire system and in the hope that the inertia can keep the system moving. If so, you should look into initial condition rather than boundary condition.

@GiudGiud if you do that you are essentially sending a shockwave through the boundaries.

**Reply by chunhuizhao478** [reply]
Thanks Gary, I think that's what I want to achieve, I'm not sure whether a constant velocity can be set with IC system, could you point out if there is a workaround?

**Reply by hugary1995** [reply, marked as answer]
It depends on your time integration scheme. You can back up the correct u_0 that gives you the desired v_0.
